Source/Scene/ShadowVolumeAppearanceShader.js

```diff
@@ -0,0 +1,347 @@
+                '    vec3 normalEC = normalize(cross(leftRight, downUp));\n' +
+                '\n';
+        }
+        return glsl;
+    }
+
+    function getLocalFunctions(shaderDependencies, planarExtents) {
+        var glsl = '';
+        if (shaderDependencies.requiresEyeCoord) {
+            glsl +=
+                'vec4 getEyeCoord(vec2 fragCoord) {\n' +
+                '    vec2 coords = fragCoord / czm_viewport.zw;\n' +
+                '    float depth = czm_unpackDepth(texture2D(czm_globeDepthTexture, coords));\n' +
+                '    vec4 windowCoord = vec4(fragCoord, depth, 1.0);\n' +
+                '    vec4 eyeCoord = czm_windowToEyeCoordinates(windowCoord);\n' +
+                '    return eyeCoord;\n' +
+                '}\n';
+        }
+        if (shaderDependencies.requiresNormalEC) {
+            glsl +=
+                'vec3 getEyeCoord3FromWindowCoord(vec2 fragCoord, float depth) {\n' +
+                '    vec4 windowCoord = vec4(fragCoord, depth, 1.0);\n' +
+                '    vec4 eyeCoord = czm_windowToEyeCoordinates(windowCoord);\n' +
+                '    return eyeCoord.xyz / eyeCoord.w;\n' +
+                '}\n' +
+
+                'vec3 getVectorFromOffset(vec4 eyeCoord, vec2 fragCoord2, vec2 positiveOffset) {\n' +
+                '    // Sample depths at both offset and negative offset\n' +
+                '    float upOrRightDepth = czm_unpackDepth(texture2D(czm_globeDepthTexture, (fragCoord2 + positiveOffset) / czm_viewport.zw));\n' +
+                '    float downOrLeftDepth = czm_unpackDepth(texture2D(czm_globeDepthTexture, (fragCoord2 - positiveOffset) / czm_viewport.zw));\n' +
+                '    // Explicitly evaluate both paths\n' +
+                '    bvec2 upOrRightInBounds = lessThan(fragCoord2 + positiveOffset, czm_viewport.zw);\n' +
+                '    float useUpOrRight = float(upOrRightDepth > 0.0 && upOrRightInBounds.x && upOrRightInBounds.y);\n' +
+                '    float useDownOrLeft = float(useUpOrRight == 0.0);\n' +
+
+                '    vec3 upOrRightEC = getEyeCoord3FromWindowCoord(fragCoord2 + positiveOffset, upOrRightDepth);\n' +
+                '    vec3 downOrLeftEC = getEyeCoord3FromWindowCoord(fragCoord2 - positiveOffset, downOrLeftDepth);\n' +
+
+                '    return (upOrRightEC - (eyeCoord.xyz / eyeCoord.w)) * useUpOrRight + ((eyeCoord.xyz / eyeCoord.w) - downOrLeftEC) * useDownOrLeft;\n' +
+                '}\n';
+        }
+        if (shaderDependencies.requiresTexcoords && planarExtents) {
+            glsl +=
+                'float computePlanarTexcoord(vec4 plane, vec3 eyeCoords, float inverseExtent) {\n' +
+                '    return (dot(plane.xyz, eyeCoords) + plane.w) * inverseExtent;\n' +
+                '}\n';
+        }
+        return glsl;
+    }
+
+    /**
+     * Tracks shader dependencies.
+     * @private
+     */
+    function ShaderDependencies() {
+        this._requiresEyeCoord = false;
```

Comment:
If I am following correctly here, for naming, you may be better off with `requiresEC` and `requiresWC` to be consistent with `UniformState` - or be completely explicit with `requiresEyeCoordinates` and `requiresWorldCoordinates`.  In general, we don't want to introduce a third set of abbreviations for us to remember.